tests/test_client.py

```diff
@@ -0,0 +1,244 @@
+import string
+
+# from requests.exceptions import HTTPError
+# from requests.auth import HTTPBasicAuth
+
+# TODO try below syncronous interaction or go all http?
+# from facebookads.objects import Ad, AdAccount, AdSet, Campaign
+
+from tap_tester.logger import LOGGER
+
+
+class TestClient():
+    # def __init__(self, config):  # TODO move to dynamic config model?
+    def __init__(self):
+        # pass in config above and get() from it or hard code?
+        self.base_url  = 'https://graph.facebook.com/'
+        self.api_version = 'v18.0'
+        self.account_id = os.getenv('TAP_FACEBOOK_ACCOUNT_ID')
+        self.access_token  = os.getenv('TAP_FACEBOOK_ACCESS_TOKEN')
+        self.account_url = self.base_url + self.api_version +'/act_{}'.format(self.account_id)
+
+        self.stream_endpoint_map = {'ads': '/ads',
+                                    'adsets': '/adsets',
+                                    'adcreative': '/adcreatives',
+                                    'ads_insights': '/insights',
+                                    'campaigns': '/campaigns',
+                                    'users': '/users',}
+
+        self.campaign_special_ad_categories = ['NONE',
+                                               'EMPLOYMENT',
+                                               'HOUSING',
+                                               'CREDIT',
+                                               # 'ISSUES_ELECTIONS_POLITICS',  # acct unauthorized
+                                               'ONLINE_GAMBLING_AND_GAMING']
+
+        # list of campaign objective values from fb docs below give "Invalid" error via api 18.0
+        # 'APP_INSTALLS', 'BRAND_AWARENESS', 'CONVERSIONS', 'EVENT_RESPONSES', 'LEAD_GENERATION',
+        # 'LINK_CLICKS', 'MESSAGES', 'OFFER_CLAIMS', 'PAGE_LIKES', 'POST_ENGAGEMENT',
+        # 'PRODUCT_CATALOG_SALES', 'REACH', 'STORE_VISITS', 'VIDEO_VIEWS'
+
+        # LOCAL_AWARENESS gives deprecated error, use REACH (reach is invalid from above)
+
+        # valid and verified objectives listed below, objectives above should be re-mapped to these
+        self.campaign_objectives = ['OUTCOME_APP_PROMOTION',
+                                    'OUTCOME_AWARENESS',
+                                    'OUTCOME_ENGAGEMENT',
+                                    'OUTCOME_LEADS',
+                                    'OUTCOME_SALES',
+                                    'OUTCOME_TRAFFIC']
+
+    def get_account_objects(self, stream):
+        assert stream in  self.stream_endpoint_map.keys(), \
+            f'Endpoint undefiend for specified stream: {stream}'
+        endpoint = self.stream_endpoint_map[stream]
+        url = self.account_url + endpoint
+        params = {'access_token': self.access_token,
+                  'limit': 100}
+        LOGGER.info(f"Getting url: {url}")
+        response = requests.get(url, params)
+        LOGGER.info(f"Returning get response: {response}")
+        return response.json()
```

Comment:
The best practices for the `requests` library is
```python
response = requests.get()
response.raise_for_status()
response.json()
```

Reply:
https://requests.readthedocs.io/en/latest/user/quickstart/#response-status-codes

Reply:
added the raise_for_status

Reply:
Does this mean my status code check in the pagination test is now redudnant?

Reply:
I don't think so. This client is just for the CRUD portion right?

And the tap is still the thing we are testing to see if pagination is working?